src/logger/logger.ts

```diff
@@ -28,30 +36,67 @@ export class Logger extends Console {
     };
   }
 
+  /**
+   * Converts any value to a string using `util.inspect()` for complex types.
+   *
+   * @param value - The value to convert
+   * @returns String representation of the value
+   */
   private valueToString(value: unknown): string {
     if (typeof value === 'string') {
       return value;
     }
+    return inspect(value, INSPECT_OPTIONS);
+  }
 
-    // Use Node.js built-in inspect for everything including errors
-    return inspect(value, {
-      compact: false,
-      depth: Infinity,
-    });
+  /**
+   * Truncates a message if it exceeds the maximum allowed length.
+   * Adds a suffix indicating how many characters were omitted.
+   *
+   * @param message - The message to truncate
+   * @returns Truncated message or original if within limits
+   */
+  private truncateMessage(message: string): string {
+    if (message.length > MAX_LOG_STRING_LENGTH) {
+      return `${message.substring(0, MAX_LOG_STRING_LENGTH)}... ${
+        message.length - MAX_LOG_STRING_LENGTH
+      } more characters`;
+    }
+    return message;
   }
 
-  logFn(args: unknown[], level: LogLevel): void {
-    if (this.options?.isLocalDevelopment) {
-      this.originalConsole[level](...args);
+  /**
+   * Core logging method that handles different execution contexts.
+   * On main thread logs with JSON formatting and tags in production, or plain in local development.
+   * In worker threads forwards messages to the main thread for processing.
+   *
+   * @param args - Values to log (converted to strings unless skipSanitization is true)
+   * @param level - Log level (info, warn, error)
+   * @param skipSanitization - Skip string conversion if args are already strings
+   */
+  logFn(args: unknown[], level: LogLevel, skipSanitization = false): void {
+    let message = skipSanitization
```

Comment:
This seems dangerous to me.
I could try to log objects, but specify the `skipSanitization` as `true`, and it would cast the `object[]` to `string[]`.
I would add a check here for verifying that all objects present are strings:

```suggestion
    let message = (skipSanitization && args.every(item => typeof item === 'string'))
```